### PR TITLE
getpathurl, isdir/isfile/exits fix, querystring fix + copy stuff

### PR DESCRIPTION
**Pull request by lnielsen**
Includes previous PR.

### CODE REVIEW

File: tests/test_fs.py [hunk 1]
```diff
@@ -592,3 +632,188 @@ def test_movedir_bad(tmppath):
...
+    copydir_bad(tmppath, False)
+
+
+def test_copydir_bad_parallel(tmppath):
+    """Test copy directory."""
+    copydir_bad(tmppath, True)
+
+
+def copydir_bad(tmppath, parallel):
+    """Test move file."""
```
Comment (otron) [inline tests/test_fs.py]: Comment doesn't match test name.

File: tests/test_fs_extras.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    """Test getcontents."""
+    fs = XRootDFS(mkurl(tmppath))
+    assert fs.getcontents('data/testa.txt') == "testa.txt\n"
+    pytest.raises(ResourceNotFoundError, fs.getcontents, 'data/invalid.txt')
+
+
+def test_setcontents(tmppath):
+    """Test setcontents."""
+    fs = XRootDFS(mkurl(tmppath))
+    fs.setcontents('data/testa.txt', "mytest") == "testa.txt\n"
```
Comment (otron) [inline tests/test_fs_extras.py]: So setcontents is supposed to return the current contents of the file?
If yes, should add a comment about it. It's not immediately obvious that that's what's being tested.

File: xrootdfs/fs.py [hunk 3]
```diff
@@ -56,11 +62,26 @@ def __init__(self, url, query=None, timeout=0):
         if not is_valid_path(base_path):
             raise InvalidPathError(path=base_path)
 
+        if queryargs:
```
Comment (otron) [inline xrootdfs/fs.py]: Could add a comment here briefly explaining what's going on :P 
(e.g. "extract query args as (x, v) dict (where x is...), raise error if...)"

File: tests/test_fs_extras.py (hunk 2 above)
Comment (lnielsen) [inline tests/test_fs_extras.py]: That's a mistake :-) It doesn't return anything.

File: xrootdfs/fs.py (hunk 3 above)
Comment (lnielsen) [inline xrootdfs/fs.py]: Added some more inline comments :-)

**Pull request merged**